framework/src/meshgenerators/BreakMeshByElementGenerator.C

```diff
@@ -61,6 +76,10 @@ BreakMeshByElementGenerator::generate()
 
   Partitioner::set_node_processor_ids(*mesh);
 
+  // We need to update the global_boundary_ids, and this is faster
+  // than a full prepare_for_use()
+  boundary_info.regenerate_id_sets();
+
```

Comment:
I think to be consistent BMEG should be doing about the same thing as BBMG, e.g. be allergic to prepare_for_use to keep the previous connectivity. Which seems undesirable. 
It's fine as it is, just a thought

in fact looking at the code seems you are going  the other way